QuickTests/FunctionalTests/DefinitionsTests+ObjC.m

```diff
@@ -0,0 +1,60 @@
+#import <XCTest/XCTest.h>
+#import <Quick/Quick.h>
+#import <Nimble/Nimble.h>
+
+#import "QCKSpecRunner.h"
+
+static NSMutableArray *fetches;
+
+QuickSpecBegin(FunctionalTests_DefinitionsSpec)
+
+define(@"one", ^NSObject *{ return @1; });
+
+it(@"should evaluate and return the defined variable", ^{
+    NSNumber *one = (NSNumber *)fetch(@"one");
+    [fetches addObject:one];
+    expect(one).to(equal(@1));
+});
+
+__block NSNumber *mutatingNumber = @2;
+define(@"number", ^NSObject *{ return mutatingNumber; });
+
+it(@"should memoize the value of the first evaluation in each example", ^{
+    NSNumber *firstFetch = (NSNumber *)fetch(@"number");
+    [fetches addObject:firstFetch];
+    mutatingNumber = @3;
+    NSNumber *secondFetch = (NSNumber *)fetch(@"number");
+    [fetches addObject:secondFetch];
+    expect(secondFetch).to(equal(firstFetch));
```

Comment:
Yeah, I think I can see why this test would fail, while the Swift one would pass.

My theory: here, the block captures a pointer to an `NSNumber` object. You then mutate what that pointer points to. In the Swift test, the closure captures the primitive `Int` value, not a pointer.